docs/concept.md

```diff
@@ -1 +1,244 @@
+
+This is the point where you start looking through the wordpress documentation and whatever you use to install it on Kubernetes (Helm Chart, Static Deployment, ...) to find out with knob to turn to make it use another DB backend.
+
+With Vedette this process should become easier. Your application just needs to declare "I want something speaking mysql protocol" and Vedette will connect it to something that speaks mysql protocol.
+
+If you want to change it - no problem
+
+Just change your apps requirements to "I want something speaking mysql protocol, that does not run on my cluster!" or change the defaults for less specific requests.
+
+## Examples
+
+## API
+
+### CapabilityClaim
+
+`CapabilityClaims` define, what your application needs from a dependency and how it wants to consume the connection informations.
+
+```yaml
+apiVersion: vedette.io/v1alpha1
+kind: CapabilityClaim
+metadata:
+  name: my-wordpress-db
+spec:
+  selector:
+    matchingLabels:
+      vedette.io/protocol: postgresql
+  config:
+  - name: connection-data
+    # config keys will be available as environment variables with the given prefix.
+    # e.g. POSTGRES_HOST, POSTGRES_PORT
+    env:
+      prefix: POSTGRES_
+  credentials:
+  - name: app-user
+    # secrets will be mounted into the container file system on the given path.
+    file:
+      path: /var/secrets/database
+      
+status:
+  phase: Bound
+  credentials:
+  - name: db-admin
+    secret:
+      name: my-wordpress-db
+  endpoints:
+  - name: postgres
+    service:
+      name: my-wordpress-db
+  conditions:
+  - type: Bound
+    status: 'True'
+```
+
+### Consuming Capabilities from a Pod
+
+To make use of capabilities from your application, all you have to do is add a reference to the `CapabilityClaim` into your `Deployment` annotations:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  annotations:
+    vedette.io/claim: my-wordpress-db
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+```
+
+Vedette will automatically add needed mounts and env vars to consume the `Secrets`, `ConfigMaps`, etc that the Claim is providing and a controller will automatically update your deployment if the Claim itself is updated.
+
+For the `CapabilityClaim` above, it will look like this:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  annotations:
+    vedette.io/claim: my-wordpress-db
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+        env:
+# --- inserted by Vedette (start)
```

Comment:
I think that it would be beneficial to reference some type of managed configuration object. When capabilities change on the platform, they would not be referenced in the particular workloads, which could cause failures.